### ISSUE TITLE
Add option to rename files in transmission-remote CLI

### ISSUE THREAD

**Issue by dllud** [transmission/transmission#104]
It seems that transmission-daemon supports renaming files inside torrents since version 2.80 (check [issue #1220 in the old tracker](https://trac.transmissionbt.com/ticket/1220)). However there is no option to do so in transmission-remote CLI. The only available option is `--move`, that can only change the top-level folder for all files.

[Transmission Remote GUI](https://sourceforge.net/projects/transgui/) is able to rename single files, thus the required functionality seems to be provided by transmission-daemon RPC.

**Comment by gelraen** [comment]
Ping? Right now I've got a problem that this feature would greatly help to solve. I have a few torrents with files I'd like to rename in the same manner, and shell loop around `transmission-remote` is much nicer than the same loop around `curl` and something that would compose JSON.

**Comment by SmellyHoneybee1** [comment]
Is this why I can't rename files in my transmission done script?

https://forum.transmissionbt.com/viewtopic.php?f=2&t=19003&sid=c93f0e8631e5e88842baa2fd44bfa60b

It seems like transmission development is slowly dying, anyone planning on joining the dev team?

**Comment by lajp** [comment]
I've worked on a implementation of this on [https://github.com/lajp/transmission-remote-rename](my branch).
My implementation works but it is quite hacky/ugly. The problem seems to be that a command (such as "--move" or "--find") cannot have more than one string argument (Please do correct me if I'm wrong). Therefore in my implementation the path has to be specified with another command("--path"). Additionally I couldn't figure out a way to get the path over to the rename command so I had to create a new global string `optargs2` to hold the value of the path.

Regardless of all these problems I find my solution to be a working one and good enough for my needs and I just wanted to inform you about its existence.

**Comment by libjared** [comment]
@lajp That is cool. AFAICT your method isn't hacky at all, since the torrent-selection arg `-t` does almost the same thing [here](https://github.com/lajp/transmission-remote-rename/blob/27697d1d85fb448852e964acfc97576560bbeac4/utils/remote.c#L2349). I would however advocate for a path length longer than 128.

The ideal for me is to select files by ID like we do torrents. While that would be cool, I doubt the datastore, nor the remote server, nor the remote client support this.

**Comment by lajp** [comment]
Ok, good to know. Perhaps I'll revisit this in the near future then and see if I can get a PR together.

**Comment by libjared** [comment]
I just pushed [my fork](https://github.com/libjared/transmission/tree/rename-path). However, this is a dead end for me because (1) Remote seems to have been rewritten in C++ since then, and (2) I mistakenly thought I could use this to move files outside of subfolders.

**Comment by lajp** [comment]
Yes, all of it has been rewritten in C++ which I personally dislike. About the usage: the feature itself is ofc not implemented in the remote, just the support for using it. I advise you to look into the daemon code.

I'll return to this matter in a couple of hours and see if I can get something compatible with the current C++ version done.

**Comment by RobFog** [comment]
Is this ready to be closed with the merge of #2905?

**Comment by lajp** [comment]
Yes, I think it should be.

**Comment by rdbisme** [comment]
Hello, it's not clear to me how to use that option? 

`transmission-remite -t <id> --rename`

What do I need to provide after rename? I tried old, new name. new, old name. Just one name. I always get "Invalid Argument"

**Comment by lajp** [comment]
Hi, the way rename is currently implemented requires the original path name that is to be renamed.

For example, if the torrent contains files `a/b`, `a/c` and you'd like to rename `a` -> `d` then you'd use the command
```sh
transmission-remote -t <id> --path a --rename d
```
